### PR TITLE
Albert Zhang assignment 3

### PR DESCRIPTION
**Pull request by AlbeartoZ-hang**

### CODE REVIEW

File: Assignment3.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Albert Zhang
+1)Not sure if this is right please correct me if I'm wrong.
+	1. O(n) = n
+	2. O(n) = n + n^2
+	3. O(n) = log(n)
+2)The printf line is not valid because the syntax is wrong. There whould also be a default case in case the number inputted its not between 1 and 4.
+3)I think this code snippet isn't valid because the i variable isn't defined as a number yet.
```
Comment (oldclesleycode) [inline Assignment3.txt:7]: right, but if you do that beforehand, it's perfectly valid!

File: Assignment3.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Albert Zhang
+1)Not sure if this is right please correct me if I'm wrong.
+	1. O(n) = n
```
Comment (oldclesleycode) [inline Assignment3.txt:3]: Right!

File: Assignment3.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+	2. O(n) = n + n^2
```
Comment (oldclesleycode) [inline Assignment3.txt:4]: Close! since it's big O notation, you don't have to care about the lower terms, so this just becomes O(n^2).

File: Assignment3.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+	3. O(n) = log(n)
```
Comment (oldclesleycode) [inline Assignment3.txt:5]: The first for loop gives a time complexity of n since you iterate through n elements. The second for loop would normally also be n, but since it multiplies the j by two each time, that means you're eliminating half the number of elements each time - this causes a time complexity of log (n). And since they're nested, you multiply the two with each other so that the time complexity is O(nlog(n)).

File: Ccount.c [hunk 5]
```diff
@@ -0,0 +1,20 @@
+/* Albert Zhang
+   Counts cs*/
+#include <stdio.h>
```
Comment (oldclesleycode) [inline Ccount.c:3]: Perfect job! 💯

File: posnums.c [hunk 6]
```diff
@@ -0,0 +1,70 @@
+/* Albert Zhang
+   Averages two numbers, then checks if they're prime and relatively prime */
```
Comment (oldclesleycode) [inline posnums.c:2]: Doesn't always print whether the numbers are relatively prime! for example, 100 and 10. Test your code more thoroughly to find these edge cases!

File: powers.c [hunk 7]
```diff
@@ -0,0 +1,28 @@
+/* Albert Zhang
+   Prints powers of 2 up to the user's input */
```
Comment (oldclesleycode) [inline powers.c:2]: you misunderstood the assignment - n isn't the number being input. The number being input is the number that 2^n should be less than. you also forgot 2^0 = 1.
